### ISSUE TITLE
"Non-fatal Exception Bad state: Future already completed" - multiple reports in Crashlytics 

### ISSUE THREAD

**Issue by egonbeermat** [bluefireteam/audioplayers#1642]
### Checklist

- [X] I read the [troubleshooting guide](https://github.com/bluefireteam/audioplayers/blob/main/troubleshooting.md) before raising this issue
- [X] I made sure that the issue I am raising doesn't already exist

### Current bug behaviour

I am seeing multiple Crashlytics reports from both iOS and Android versions of my app:

```
Non-fatal Exception: io.flutter.plugins.firebase.crashlytics.FlutterError: Bad state: Future already completed
       at _AsyncCompleter.complete(dart:async)
       at AudioPlayer._completePrepared.<fn>(audioplayer.dart:307)
```

No further stack trace information is presented in any of the reports

### Expected behaviour

Not see these error reports :)

### Steps to reproduce

Unfortunately, I cannot reproduce this locally during my own testing, so this is probably just an FYI for you guys...

my application release history relevant to when these errors appeard:

myApp version 2.1.0 - Flutter 3.7.6, audioplayers 3.0.1 -> no reports
myApp version 2.1.2 - Flutter 3.10.6, audioplayers 5.1.0 -> error reports received
myApp version 2.1.3 - Flutter 3.13.2, audioplayers 5.0.0 -> error reports received

### Affected platforms

Android, iOS

### Platform details

Android:  7, 9, 12 and 13
iOS:  15.5.0, 15.7.8, 16.1.1, 16.5.1, 16.2.0, 16.3.1, 16.4.1, 16.6.0, 16.6.1

### AudioPlayers Version

5.0.0, 5.1.0

### Build mode

release

### Logs

<details>
  <summary> Flutter doctor</summary>

  ```
[✓] Flutter (Channel stable, 3.13.2, on macOS 13.4.1 22F770820d darwin-arm64, locale en-US)
    • Flutter version 3.13.2 on channel stable at /Users/ItBeMe/Documents/Development/FlutterSDK/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ff5b5b5fa6 (3 weeks ago), 2023-08-24 08:12:28 -0500
    • Engine revision b20183e040
    • Dart version 3.1.0
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Volumes/Samsung960Pro/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • ANDROID_HOME = /Volumes/Samsung960Pro/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.82.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.72.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.4.1 22F770820d darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 116.0.5845.187

[✓] Network resources
    • All expected network resources are available.

• No issues found!
  ```
</details>

### Working on PR

no way

**Comment by Gustl22** [comment]
This is probably fixed by #1660, anyways there should be an error beforehand, that the source couldn't be loaded as its not supported or sth else, see https://github.com/bluefireteam/audioplayers/blob/main/troubleshooting.md.

**Comment by egonbeermat** [comment]
Thanks. There is not much else to add from my end - the app only plays audio from /assets, and they all work locally on devices and simulators that match some of the devices reporting this in Crashlytics. As there is no further data in the Crashlytics reports, or any related error reports coming in around the same time, and that this is not reproducible locally, theres not much to add and nothing useful for you to work with, which is why I reported it as an FYI :)
